app/src/main/java/com/islamversity/reyan/service/MyFirebaseMessagingService.kt

```diff
@@ -0,0 +1,101 @@
+package com.islamversity.reyan.service
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.media.RingtoneManager
+import android.os.Build
+import android.os.Bundle
+import androidx.core.app.NotificationCompat
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import com.islamversity.reyan.MainActivity
+import com.islamversity.reyan.R
+import com.islamversity.reyan.service.NotificationDataType.Companion.NOTIFICATION_DATA_KEY
+import timber.log.Timber
+
+
+class MyFirebaseMessagingService: FirebaseMessagingService() {
+
+
+    override fun onNewToken(token: String) {
+        Timber.d("Notification : Refreshed token: $token")
+        sendFCMTokenToServer(token)
+    }
+
+    private fun sendFCMTokenToServer(token: String) {
+        // Cache.set(CacheKeys.Firebase.FCM_TOKEN_FIREBASE, token)
+        Timber.d("Notification : sendRegistrationTokenToServer($token)")
+    }
+
+    override fun onMessageReceived(remoteMessage: RemoteMessage) {
+        Timber.d("Notification : From: ${remoteMessage.from}")
+
+        val data  = remoteMessage.data
+
+        // Check if message contains a data payload.
+        if (data.isNotEmpty()) {
+            Timber.d("Notification : Message data payload: $data")
+
+            val bundle = Bundle()
+            for ((key, value) in data.entries) {
+                bundle.putString(key, value)
+            }
+
+            showNotification(
+                data["title"],
+                data["body"],
+                bundle
+            )
```

Comment:
I think we can use `Factory` patter for our different types of notifications

Reply:
What do u mean? we do not  handle notificatiion types here.

Reply:
yeah I know
that's why we can create factory and pass the type to it so that factory can create the required notification for us
that in case of new notification types all we need to do is just to create a new notification in our factory
nothing will be changed in here

Reply:
do we have different notification to show based on type ?

Reply:
not now
but will definitely have some